indexing-service/src/main/java/io/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java

```diff
@@ -1335,11 +1336,11 @@ void taskAddedOrUpdated(final TaskAnnouncement announcement, final WorkerHolder
         String taskId,
         Worker worker,
         TaskLocation location,
-        Task task,
+        @Nullable Task task,
         State state
     )
     {
-      super(taskId, worker, location);
+      super(taskId, task == null ? null : task.getType(), worker, location);
```

Comment:
taskType should be updated in setTask(Task) method as well.

Reply:
Thanks. Added.